sympy/core/function.py

```diff
@@ -2824,19 +2824,95 @@ def _eval_expand_hint(self, **hints):
 
 # This is a special application of two hints
 
-def _mexpand(expr, recursive=False):
-    # expand multinomials and then expand products; this may not always
-    # be sufficient to give a fully expanded expression (see
-    # test_issue_8247_8354 in test_arit)
-    if expr is None:
-        return
-    was = None
-    while was != expr:
-        was, expr = expr, expand_mul(expand_multinomial(expr))
-        if not recursive:
-            break
-    return expr
+def mexpand(expr, recursive=False, *, numer=None, denom=None, frac=None):
+    """expand multinomials and then expand products in ``expr``.
+
+    If recursive is False, this may not always be sufficient to
+    give a fully expanded expression.
+
+    To target expansion only on the numerator or denominator, set
+    ``numer`` or ``denom`` to True, respectively. To keep numerator
+    and denominator over each other and expand each, make `frac`` True.
+
+    EXAMPLES
+    ========
+
+    >>> from sympy.core.function import mexpand
+    >>> from sympy.abc import x
+    >>> eq = (x - 1)**2/(1 - x)**2
+    >>> mexpand(eq, frac=True)
+    1
+    >>> mexpand(eq, numer=True)
+    (x**2 - 2*x + 1)/(1 - x)**2
+    >>> mexpand(eq, denom=True)
+    (x - 1)**2/(x**2 - 2*x + 1)
+    >>> mexpand(eq)
+    x**2/(x**2 - 2*x + 1) - 2*x/(x**2 - 2*x + 1) + 1/(x**2 - 2*x + 1)
+
+    >>> eq -= 1
+    >>> mexpand(eq, frac=True)
+    0
+    >>> mexpand(eq, numer=True)
+    0
+    >>> mexpand(eq, denom=True)
+    -1 + (x - 1)**2/(1 - x)**2
+    >>> mexpand(eq)
+    0
 
+    See Also
+    ========
+    mexpand_cse
+    """
+    from sympy.simplify.radsimp import fraction
+    def do(expr):
+        was = None
+        while was != expr:
+            was, expr = expr, expand_mul(expand_multinomial(expr))
+            if not recursive:
+                break
+        return expr
+    n, d = fraction(sympify(expr), exact=True)
+    both = False
+    if frac or numer is None and denom is None:
+        both = numer = denom = True
+    if numer:
+        n = do(n)
+    if denom:
+        if not d.is_Atom:
+            d = do(d)
+    if n.is_Add and both and not frac:
+        return n.func(*[i/d for i in n.args])
+    return n/d
+
+
+def mexpand_cse(e, _final_denom=True):
+    """return a recursively expanded expression using
+    multinomial and product expansions, but introduce
+    any repeated subexpressions 1 at a time while doing so.
+    Unlike `mexpand`, the results here will always be
+    expressed as a rational expression (but common factors
+    will not necessarily be automatically be cancelled).
+
+    EXAMPLES
+    ========
```

Comment:
The one line summary should be more like one line at the start of the docstring and EXAMPLEs should be capitalised like Examples.

Reply:
That's where it is. Notice that this is the `mexpand_cse` function, not the `mexpand` function.